walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,54 @@
 
 package com.google.sps;
 
+import java.util.List;
+import java.util.Arrays;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // if no attendees, return all day
+    if (request.getAttendees().isEmpty()) return Arrays.asList(TimeRange.WHOLE_DAY);
+
+    Collection<TimeRange> availableTimes = new ArrayList<TimeRange>();
+
+    // if request is longer than a day, return no events
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) return availableTimes;
+
+    // make a sorted list of the events
+    List<Event> eventList = new ArrayList(events);
+    Collections.sort(eventList, (Event e1, Event e2) -> TimeRange.ORDER_BY_START.compare(e1.getWhen(), e2.getWhen()));
```

Comment:
Are we able to assume that these events do not overlap?

Reply:
I don't think we can